### PR TITLE
Show dialog message if URL is malformed.

### LINKED ISSUE #1616: Malformed URL fragment does not give any feedback except an empty canvas
For example: a truncated URL causes an "unexpected EOF" error during decompression.

Suggested fix:
- catch decoding/deserialize error, show dialog saying something like "The URL is malformed: the UI settings could not be restored"

### PR DESCRIPTION
**Pull request by ollimeier**
Fixes #1616

### CODE REVIEW

File: src/fontra/client/core/utils.js [hunk 1]
```diff
@@ -515,7 +515,11 @@ export function loadURLFragment(fragment) {
   if (fragment[0] != "#") {
     throw new Error("assert -- invalid fragment");
   }
-  return JSON.parse(strFromU8(unzlibSync(base64ToBytes(fragment.slice(1)))));
+  try {
+    return JSON.parse(strFromU8(unzlibSync(base64ToBytes(fragment.slice(1)))));
+  } catch {
+    return false;
```
Comment (justvanrossum) [inline src/fontra/client/core/utils.js:521]:
```suggestion
    return null;
```

**Review by justvanrossum — COMMENTED** [review]
One small comment. Otherwise looks fine.

**Review by justvanrossum — APPROVED** [review]

**Review by justvanrossum — CHANGES_REQUESTED** [review]
I just now tried it myself, but while there is a dialog now, there is a different error after this, prevnting the URL to be updated later:

image

Paste a malfored URL in a new tab, dismiss the dialog, then add text to the text entry field. Notice how the URL does not get updated: it stays at the malformed URL.

**Comment by ollimeier** [comment]
> I just now tried it myself, but while there is a dialog now, there is a different error after this, prevnting the URL to be updated later:
> 
> image
> Paste a malfored URL in a new tab, dismiss the dialog, then add text to the text entry field. Notice how the URL does not get updated: it stays at the malformed URL.

Thanks for finding this. I fixed it.

**Review by justvanrossum — APPROVED** [review]

**Pull request merged**